src/hotspot/share/gc/shared/gcArguments.cpp

```diff
@@ -126,7 +129,14 @@ void GCArguments::initialize_heap_flags_and_sizes() {
       vm_exit_during_initialization("Incompatible minimum and maximum heap sizes specified");
     }
   }
-
+#if defined(LINUX)
+  if (OSContainer::is_containerized() && FLAG_IS_CMDLINE(InitialHeapSize)) {
+    jlong memswBytes = OSContainer::memory_and_swap_limit_in_bytes();
+    if ((memswBytes > 0) && (InitialHeapSize >= (julong) memswBytes)) {
+      vm_exit_during_initialization("Initial heap size set to a larger value than the container memory & swap limit");
+    }
+  }
+#endif
```

Comment:
The current code is structured with carefully added newlines to aid readability. This patch removes one of those newlines, and replace it with this block. Could you keep the current format and add code as a separate block with newlines before and after? Maybe also consider moving this to after the more primitive checks in 140-149?